core/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateAction.java

```diff
@@ -156,6 +160,16 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
                 }
             );
         }
+
+        private void handleShardFailureOnMaster(final ShardRoutingEntry shardRoutingEntry, ClusterStateTaskListener listener) {
```

Comment:
also, I wonder if we should just inline this method.. not sure it buys us much now.